staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -426,11 +426,36 @@ func (g *Cloud) AddAliasToInstance(nodeName types.NodeName, alias *net.IPNet) er
 // Gets the named instances, returning cloudprovider.InstanceNotFound if any
 // instance is not found
 func (g *Cloud) getInstancesByNames(names []string) ([]*gceInstance, error) {
+	foundInstances, err := g.getFoundInstanceByNames(names)
+	if err != nil {
+		return nil, err
+	}
+	var failed []string
+	foundInstanceNames := sets.NewString()
+	for _, instance := range foundInstances {
+		foundInstanceNames.Insert(instance.Name)
+	}
```

Comment:
I don't understand the reason for this - the contract of getFoundINstanceByNames is saying that it returns exactly the map with entry for every name from given names and no other entries.

What's the point of having this one?

Reply:
We changed it so it just returns a lost of gceinstances instead. This loop is used to calculate the diff between the expected and found entries so it can be logged.

Reply:
But it's NOT what the function is doing - the function getFoundInstanceByNames can ONLY return instances that are in the list of passed names, not other names. Lines:467 and 488.

We don't want to check the contract of the method here - it's unnecessary calculations and unnecessary waste of resources.

Reply:
If not sure I follow - the loop here and below is to find the instances that getInstancesByName expected to get but didn't actually get. This is because if getFoundInstancesByName can't find an instance for a name, it just quietly doesn't include it in the returned slice. It's not about checking for extra instances returned.

Reply:
I see now - you changed the interface of the method since yesterday - it's no longer returning InstanceOrError.

While I preferred the previous interface, this is private one, and I don't think it's worth changing.